explorer/interpreter/resolve_unformed.cpp

```diff
@@ -117,15 +123,35 @@ static auto ResolveUnformed(Nonnull<const Expression*> expression,
       }
       break;
     case ExpressionKind::SimpleMemberAccessExpression:
-      CARBON_RETURN_IF_ERROR(ResolveUnformed(
-          &cast<SimpleMemberAccessExpression>(*expression).object(), flow_facts,
-          FlowFacts::ActionType::Check));
+    case ExpressionKind::CompoundMemberAccessExpression:
+    case ExpressionKind::BaseAccessExpression:
+      CARBON_RETURN_IF_ERROR(
+          ResolveUnformed(&cast<MemberAccessExpression>(*expression).object(),
+                          flow_facts, FlowFacts::ActionType::Check));
       break;
     case ExpressionKind::BuiltinConvertExpression:
       CARBON_RETURN_IF_ERROR(ResolveUnformed(
           cast<BuiltinConvertExpression>(*expression).source_expression(),
           flow_facts, FlowFacts::ActionType::Check));
       break;
+    case ExpressionKind::IndexExpression:
+      CARBON_RETURN_IF_ERROR(
+          ResolveUnformed(&cast<IndexExpression>(*expression).object(),
+                          flow_facts, FlowFacts::ActionType::Check));
+      CARBON_RETURN_IF_ERROR(
+          ResolveUnformed(&cast<IndexExpression>(*expression).offset(),
+                          flow_facts, FlowFacts::ActionType::Check));
+      break;
```

Comment:
```suggestion
    case ExpressionKind::IndexExpression: {
      const auto& index_exp = cast<IndexExpression>(*expression);
      CARBON_RETURN_IF_ERROR(
          ResolveUnformed(&index_exp.object(),
                          flow_facts, FlowFacts::ActionType::Check));
      CARBON_RETURN_IF_ERROR(
          ResolveUnformed(&index_exp.offset(),
                          flow_facts, FlowFacts::ActionType::Check));
      break;
    }
```

It took me a few looks to spot `object` vs `offset`, that this wasn't just a copy-paste mistake. Maybe factor out the cast?

(TBH, makes me think it'd also help if we cut out a step in `FlowFacts::ActionType::Check` to reduce the duplicate text, but that seems out of scope)